### DISCUSSION TITLE
Mapping 'clone-in-kitty' or 'kitty @ launch --copy-env'?

### DISCUSSION THREAD

**Discussion by 097115** [kovidgoyal/kitty#6249] [Q&A] [answered]
Let's say I have kitty running. Now, if in this terminal I **manually** run either `clone-in-kitty` or `kitty @ launch --copy-env`, then I indeed get a newly cloned environment including vars that were set in that parent kitty, exactly as described in the documentation.

But can I somehow **map** either of these calls?

I tried something like this:

    kitty -o "allow_remote_control yes" -o "map f1 launch sh -c 'kitty @ launch --copy-env'"

or this:

    kitty -o "allow_cloning yes" -o "map f1 launch clone-in-kitty"

But both these methods fail (the former doesn't produce any visible error, just acts somehow weirdly and doesn't inherit introduced vars, while the latter fails to launch a child, reporting that the argument list is too long).

Any hints may be?

Thanks.

**Comment by kovidgoyal** [comment, marked as answer]
No you cannot. There is no way for a parent process to access the env
vars in a child process. When you run them manually, you are running
them inside the child process so they have access to its env vars. The
best you can do is map send_text to produce that text followed by \r to
execute. This mapping will work as long as the window is at an empty
shell prompt.

**Comment by 097115** [comment]
Expected this :)

Thanks, Kovid!
